lib/awful/ewmh.lua

```diff
@@ -68,6 +85,55 @@ function ewmh.activate(c, context, hints) -- luacheck: no unused args
     end
 end
 
+--- Add an activate (focus stealing) filter function.
+--
+-- The callback takes the following parameters:
+--
+-- * **c** (*client*) The client requesting the activation
+-- * **context** (*string*) The activation context.
+-- * **hints** (*table*) Some additional hints (depending on the context)
+--
+-- If the callback returns `true`, the client will be activated unless the `force`
+-- hint is set. If the callback returns `false`, the activation request is
+-- cancelled. If the callback returns `nil`, the previous callback will be
+-- executed. This will continue until either a callback handles the request or
+-- when it runs out of callbacks. In that case, the request will be granted if
+-- the client is visible.
+--
+-- For example, to block Firefox from stealing the focus, use:
+--
+--    awful.ewmh.add_activate_filter(function(c)
+--        if c.class == "Firefox" then return false end
+--    end)
+--
+-- @tparam function f The callback
+-- @see activate_filters
+-- @see remove_activate_filter
+function ewmh.add_activate_filter(f)
+    table.insert(ewmh.activate_filters, f)
+end
+
+--- Remove an activate (focus stealing) filter function.
+-- This is an helper to avoid dealing with `ewmh.add_activate_filter` directly.
+-- @tparam function f The callback
+-- @treturn boolean If the callback existed
+-- @see activate_filters
+-- @see add_activate_filter
+function ewmh.remove_activate_filter(f)
+    for k, v in ipairs(ewmh.activate_filters) do
+        if v == f then
+            table.remove(ewmh.activate_filters, k)
+
+            -- In case the callback is there multiple time.
+            ewmh.remove_activate_filter(f)
```

Comment:
Is the order of these filters really relevant? I'd expect them to be side-effect-free aka "their order is relevant". If so, I'd simplify the filter management:
```lua
function add(f)
  ewmh.activate_filters[t] = true
end
function remove(f)
  local old = ewmh.activate_filters[t]
  ewmh.activate_filters[t] = nil
  return old == true
end
```